irohad/ametsuchi/impl/postgres_command_executor.hpp

```diff
@@ -18,19 +18,23 @@ namespace shared_model {
 namespace iroha {
   namespace ametsuchi {
 
-    class PostgresCommandExecutor : public CommandExecutor {
+    class PostgresCommandExecutor final : public CommandExecutor {
      public:
       PostgresCommandExecutor(
-          soci::session &transaction,
+          std::unique_ptr<soci::session> sql,
           std::shared_ptr<shared_model::interface::PermissionToString>
               perm_converter);
 
+      ~PostgresCommandExecutor();
```

Comment:
Could be removed I guess.

Reply:
It is tricky, but it is needed because of an undefined (forward-declared) member class. As I understand the situation, such a header file does not provide enough info to destroy `PostgresCommandExecutor`, and we need to promise that we have a destructor that will do it. And the destructor is generated after the member class is defined (in cpp).